### ISSUE TITLE
Remote debug doesn't work

### ISSUE THREAD

**Issue by luisxiaomai** [openstf/stf#170]
Hi guys,

I connected to a device and see "10.xx.xx.xx:7409" in remote debug tab, and I run "adb connect 10.xx.xx.xx:7409". It says "connected to 10.xx.xx.xx:7409", but I run "adb devices" I can't see this device. I also can't find it in my android studio? 

Please give some help, thanks.

B.R.
Luis

**Comment by sorccu** [comment]
Is `stf local` running on the same machine as Android Studio?

Have you confirmed your ADB key in the STF UI?

**Comment by luisxiaomai** [comment]
STF local running in one machine. 
One user login from another machine to connect this device and do the remote debug. Then he stop using this machine.(ADB key not removed?)
Another user login from third machine to connect this device  and want to debug, but after he run "adb connect 10.xx.xx.xx:7409", but key pops up and can't find this device.
But I return to the first user and remove the key, use second to login again and run "adb connect 10.xx.xx.xx:7409", key pops up and it works.

Can you help and fix this issue?

**Comment by sorccu** [comment]
Do both users have unique ADB keys? Are both of them verified?

**Comment by luisxiaomai** [comment]
Second user don't have key, no add keys dialog pops up after he run "adb connect 10.xx.xx.xx:7409".

**Comment by sorccu** [comment]
Does his computer work with normal remote adb ('adb tcpip 5555' & adb
connect ip-of-device)?

**Comment by vbanthia** [comment]
@sorccu 
Currently, two users cannot have same ADB keys. But I think we should allow users to share keys.

A valid use case is when you will use STF APIs for test automation. For example, if you are running test from Jenkins slave using access token that Jenkins slave machine can be shared by many users.

**Comment by sorccu** [comment]
Allow how? How would we know who's trying to use it then?

**Comment by vbanthia** [comment]
yeah, that we will need to figure it out. But I just realised that there will be use cases where user will be sharing adb-keys. I cannot think any good solution for now, but will keep this in mind.

**Comment by sorccu** [comment]
I think GitHub deploy keys behave in a very similar way. Each deploy key for every repository must be unique, since that's the only way they can know which repository it's for. And yes, it is super, super annoying with CI, but since even GitHub hasn't been able to solve that issue, it could be that it simply cannot be solved without modifying ADB's authentication mechanism, which would require modifying the adb binaries themselves, or tunneling the whole connection somehow (which wouldn't actually be bad for security with something like openssh, but would complicate usage).

**Comment by sorccu** [comment]
GitHub can and does partially solve the issue by providing alternate ways to authenticate (like personal access tokens), but since we're stuck with what the adb binary gives us right now, we can't do the same.

**Comment by vbanthia** [comment]
yeah, this will be difficult.

And for problem with shared machine(Jenkins Slave) we can advise user to use access token of "bot" user instead of their own. And use same bot user for tests targeting on that machine.

**Comment by yuzhidi** [comment]
@sorccu 
You said , Is stf local running on the same machine as Android Studio?
stf local and Android Studio run on the same machine will cause error ?
